wkcuber/image_readers.py

```diff
@@ -199,11 +208,128 @@ def read_dtype(self, file_name: str) -> str:
             ].dtype.name
 
 
+class CziImageReader(ImageReader):
+    def __init__(self) -> None:
+        self.tile_shape = None
+
+    # returns format (X, Y)
+    def _read_array_single_channel(
+        self, czi_file: CziFile, channel_slice: int, dtype: np.dtype, z_slice: int
+    ) -> np.ndarray:
+        channel_index = czi_file.axes.find("C")
+        x_index = czi_file.axes.find("X")
+        y_index = czi_file.axes.find("Y")
+        z_index = czi_file.axes.find("Z")
+        channel_file_start = czi_file.start[channel_index]
+        z_file_start = czi_file.start[z_index]
+        for entry in czi_file.filtered_subblock_directory:
+            if (
+                entry.start[z_index] - z_file_start == z_slice
+                and entry.start[channel_index] - channel_file_start == channel_slice
+            ):
+                data = entry.data_segment().data()
+                if x_index > y_index:
+                    data = data.reshape(data.shape[y_index], data.shape[x_index])
+                    data = data.swapaxes(0, 1)
+                else:
+                    data = data.reshape(data.shape[x_index], data.shape[y_index])
```

Comment:
Can you add a comment which explains the reasoning behind this a bit?

Reply:
I have added some comments. The reasoning is pretty similar to the all channel case, but is easier because we only need to handle two dimensions.

Reply:
Great, thanks for dealing with this complexity :+1: